### ISSUE TITLE
serde_codegen fails to compile on recent nightly with -Z orbit

### ISSUE THREAD

**Issue by alexbool** [rust-lang/rust#34119]
When I compile `serde_codegen 0.7.8` with `-Z orbit` on Windows 32-bit, I get this error:
```
Assertion failed: isa<X>(Val) && "cast<Ty>() argument of incompatible type!", file C:\bot\slave\nightly-dist-rustc-win-msvc-32\build\src\llvm\include\llvm/Support/Casting.h, line 223
Build failed, waiting for other jobs to finish...
error: Could not compile `serde_codegen`.
```

Meta:
```
Alexander@VAIO-FZ13 MINGW64 ~
$ rustc -vV
rustc 1.11.0-nightly (1c975eafa 2016-06-05)
binary: rustc
commit-hash: 1c975eafa934e291e4f94b7252faae767de17313
commit-date: 2016-06-05
host: i686-pc-windows-msvc
release: 1.11.0-nightly
```

The build succeeds at least on OS X.

**Comment by retep998** [comment]
The biggest codegen difference between -msvc and other targets is the SEH exception stuff, which may be causing this, possibly.

**Comment by eddyb** [comment]
Hang on, that includes the changes which broke `syntex_syntax` and which I fixed in #34128.

@alexbool What version did you try on OSX?

**Comment by alexbool** [comment]
@eddyb version on OS X is:
```
alexbool@alexbool-osx ~/D/I/serde-codegen-failure> rustc -vV
rustc 1.11.0-nightly (1c975eafa 2016-06-05)
binary: rustc
commit-hash: 1c975eafa934e291e4f94b7252faae767de17313
commit-date: 2016-06-05
host: x86_64-apple-darwin
release: 1.11.0-nightly
```

**Comment by eddyb** [comment]
@alexbool Hmm, that's the same version, it shouldn't be able to compile `syntex_syntax` with `RUSTFLAGS=-Zorbit`.
Unless version `0.7` is old enough that it doesn't sport `struct Name(u32);` (that definition used to be `type Name = u32;` which would explain why it works).

@retep998 started a windows build of #34128, so we might find out soon, one way or another.

**Comment by alexbool** [comment]
@eddyb I am trying to make a minimal example now, seems that it isn't so easy

**Comment by eddyb** [comment]
@alexbool A backtrace or the last hundred or so lines from `RUST_LOG=rustc_trans::mir` could reveal the problem but I doubt either is possibly on the nightly.

**Comment by alexbool** [comment]
@eddyb **UPD** only happens on release builds. Minimal example on the way

**Comment by alexbool** [comment]
`Cargo.toml`
```
[package]
name = "serde-codegen-failure"
version = "0.1.0"
authors = ["Alexander Bulaev <alexbool@yandex-team.ru>"]

[dependencies]
serde = "*"
serde_macros = "*"
```

`src/lib.rs`
```rust
#![feature(custom_derive, plugin)]
#![plugin(serde_macros)]

extern crate serde;
```

`~/.cargo/config`
```
[build]
rustflags = ["-Ctarget-cpu=native", "-Zorbit"]
```
(removing `-Ctarget-cpu=native` doesn't help)

```
Alexander@VAIO-FZ13 MINGW64 ~/serde-codegen-failure (master)
$ cargo build --verbose --release
   Compiling aster v0.17.0
     Running `rustc C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\aster-0.17.0\\src\\lib.rs
--crate-name aster --crate-type lib -C opt-level=3 -C metadata=8cc5f970a844ae51 -C extra-filename=-8cc5f970a844ae51 --ou
t-dir C:\\Users\\Alexander\\serde-codegen-failure\\target\\release\\deps --emit=dep-info,link -L dependency=C:\\Users\\A
lexander\\serde-codegen-failure\\target\\release\\deps -L dependency=C:\\Users\\Alexander\\serde-codegen-failure\\target
\\release\\deps --cap-lints allow -Ctarget-cpu=native -Zorbit -Ctarget-cpu=native -Zorbit`
   Compiling quasi v0.11.0
     Running `rustc C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\quasi-0.11.0\\src\\lib.rs
--crate-name quasi --crate-type lib -C opt-level=3 -C metadata=f0f90e4a9a9f69ab -C extra-filename=-f0f90e4a9a9f69ab --ou
t-dir C:\\Users\\Alexander\\serde-codegen-failure\\target\\release\\deps --emit=dep-info,link -L dependency=C:\\Users\\A
lexander\\serde-codegen-failure\\target\\release\\deps -L dependency=C:\\Users\\Alexander\\serde-codegen-failure\\target
\\release\\deps --cap-lints allow -Ctarget-cpu=native -Zorbit -Ctarget-cpu=native -Zorbit`
   Compiling serde v0.7.8
     Running `rustc C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\serde-0.7.8\\src\\lib.rs -
-crate-name serde --crate-type lib -C opt-level=3 --cfg feature=\"std\" --cfg feature=\"default\" -C metadata=90f0afafc2
b6e513 -C extra-filename=-90f0afafc2b6e513 --out-dir C:\\Users\\Alexander\\serde-codegen-failure\\target\\release\\deps
--emit=dep-info,link -L dependency=C:\\Users\\Alexander\\serde-codegen-failure\\target\\release\\deps -L dependency=C:\\
Users\\Alexander\\serde-codegen-failure\\target\\release\\deps --cap-lints allow -Ctarget-cpu=native -Zorbit -Ctarget-cp
u=native -Zorbit`
   Compiling serde_codegen v0.7.8
     Running `rustc C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\serde_codegen-0.7.8\\build
.rs --crate-name build_script_build --crate-type bin -C opt-level=3 --cfg feature=\"quasi_macros\" --cfg feature=\"night
ly\" --out-dir C:\\Users\\Alexander\\serde-codegen-failure\\target\\release\\build\\serde_codegen-9f41a709c17aa9aa --emi
t=dep-info,link -L dependency=C:\\Users\\Alexander\\serde-codegen-failure\\target\\release\\deps -L dependency=C:\\Users
\\Alexander\\serde-codegen-failure\\target\\release\\deps --cap-lints allow -Ctarget-cpu=native -Zorbit -Ctarget-cpu=nat
ive -Zorbit`
     Running `C:\Users\Alexander\serde-codegen-failure\target\release\build\serde_codegen-9f41a709c17aa9aa\build-script-
build`
   Compiling quasi_codegen v0.11.0
     Running `rustc C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\quasi_codegen-0.11.0\\src\
\lib.rs --crate-name quasi_codegen --crate-type lib -C opt-level=3 -C metadata=d15a779fc360468f -C extra-filename=-d15a7
79fc360468f --out-dir C:\\Users\\Alexander\\serde-codegen-failure\\target\\release\\deps --emit=dep-info,link -L depende
ncy=C:\\Users\\Alexander\\serde-codegen-failure\\target\\release\\deps -L dependency=C:\\Users\\Alexander\\serde-codegen
-failure\\target\\release\\deps --extern aster=C:\\Users\\Alexander\\serde-codegen-failure\\target\\release\\deps\\libas
ter-8cc5f970a844ae51.rlib --cap-lints allow -Ctarget-cpu=native -Zorbit -Ctarget-cpu=native -Zorbit`
   Compiling quasi_macros v0.11.0
     Running `rustc C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\quasi_macros-0.11.0\\src\\
lib.rs --crate-name quasi_macros --crate-type dylib -C prefer-dynamic -C opt-level=3 -C metadata=30f11f7edf6a7bdc -C ext
ra-filename=-30f11f7edf6a7bdc --out-dir C:\\Users\\Alexander\\serde-codegen-failure\\target\\release\\deps --emit=dep-in
fo,link -L dependency=C:\\Users\\Alexander\\serde-codegen-failure\\target\\release\\deps -L dependency=C:\\Users\\Alexan
der\\serde-codegen-failure\\target\\release\\deps --extern quasi_codegen=C:\\Users\\Alexander\\serde-codegen-failure\\ta
rget\\release\\deps\\libquasi_codegen-d15a779fc360468f.rlib --cap-lints allow -Ctarget-cpu=native -Zorbit -Ctarget-cpu=n
ative -Zorbit`
     Running `rustc C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\serde_codegen-0.7.8\\src\\
lib.rs --crate-name serde_codegen --crate-type lib -C opt-level=3 --cfg feature=\"quasi_macros\" --cfg feature=\"nightly
\" -C metadata=17040f1a2091a93f -C extra-filename=-17040f1a2091a93f --out-dir C:\\Users\\Alexander\\serde-codegen-failur
e\\target\\release\\deps --emit=dep-info,link -L dependency=C:\\Users\\Alexander\\serde-codegen-failure\\target\\release
\\deps -L dependency=C:\\Users\\Alexander\\serde-codegen-failure\\target\\release\\deps --extern quasi=C:\\Users\\Alexan
der\\serde-codegen-failure\\target\\release\\deps\\libquasi-f0f90e4a9a9f69ab.rlib --extern quasi_macros=C:\\Users\\Alexa
nder\\serde-codegen-failure\\target\\release\\deps\\quasi_macros-30f11f7edf6a7bdc.dll --extern aster=C:\\Users\\Alexande
r\\serde-codegen-failure\\target\\release\\deps\\libaster-8cc5f970a844ae51.rlib --cap-lints allow -Ctarget-cpu=native -Z
orbit -Ctarget-cpu=native -Zorbit`
Assertion failed: isa<X>(Val) && "cast<Ty>() argument of incompatible type!", file C:\bot\slave\nightly-dist-rustc-win-m
svc-32\build\src\llvm\include\llvm/Support/Casting.h, line 223
error: Could not compile `serde_codegen`.

Caused by:
  Process didn't exit successfully: `rustc C:\Users\Alexander\.cargo\registry\src\github.com-1ecc6299db9ec823\serde_code
gen-0.7.8\src\lib.rs --crate-name serde_codegen --crate-type lib -C opt-level=3 --cfg feature="quasi_macros" --cfg featu
re="nightly" -C metadata=17040f1a2091a93f -C extra-filename=-17040f1a2091a93f --out-dir C:\Users\Alexander\serde-codegen
-failure\target\release\deps --emit=dep-info,link -L dependency=C:\Users\Alexander\serde-codegen-failure\target\release\
deps -L dependency=C:\Users\Alexander\serde-codegen-failure\target\release\deps --extern quasi=C:\Users\Alexander\serde-
codegen-failure\target\release\deps\libquasi-f0f90e4a9a9f69ab.rlib --extern quasi_macros=C:\Users\Alexander\serde-codege
n-failure\target\release\deps\quasi_macros-30f11f7edf6a7bdc.dll --extern aster=C:\Users\Alexander\serde-codegen-failure\
target\release\deps\libaster-8cc5f970a844ae51.rlib --cap-lints allow -Ctarget-cpu=native -Zorbit -Ctarget-cpu=native -Zo
rbit` (exit code: 3221226505)

```

**Comment by alexbool** [comment]
Also build options (`-Ctarget-cpu=native -Zorbit -Ctarget-cpu=native -Zorbit`) are somehow passed twice, seems like there's also a cargo bug

**Comment by eddyb** [comment]
@alexbool Isn't it easier to just run `RUSTFLAGS=-Zorbit cargo build` in `serde_codegen`'s source dir?

**Comment by alexbool** [comment]
@eddyb Yep, something like this works:
```
Alexander@VAIO-FZ13 MINGW64 ~/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_codegen-0.7.8
$ cargo build --release --verbose --no-default-features --features "nightly"
   Compiling aster v0.17.0
     Running `rustc C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\aster-0.17.0\\src\\lib.rs
--crate-name aster --crate-type lib -C opt-level=3 -C metadata=8cc5f970a844ae51 -C extra-filename=-8cc5f970a844ae51 --ou
t-dir C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\serde_codegen-0.7.8\\target\\release\\de
ps --emit=dep-info,link -L dependency=C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\serde_co
degen-0.7.8\\target\\release\\deps -L dependency=C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec82
3\\serde_codegen-0.7.8\\target\\release\\deps --cap-lints allow -Ctarget-cpu=native -Zorbit`
   Compiling serde_codegen v0.7.8 (file:///C:/Users/Alexander/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_code
gen-0.7.8)
     Running `rustc build.rs --crate-name build_script_build --crate-type bin -C opt-level=3 --cfg feature=\"nightly\" -
-cfg feature=\"quasi_macros\" --out-dir C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\serde_
codegen-0.7.8\\target\\release\\build\\serde_codegen-da094a7fbaf20dd5 --emit=dep-info,link -L dependency=C:\\Users\\Alex
ander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\serde_codegen-0.7.8\\target\\release -L dependency=C:\\Users\
\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\serde_codegen-0.7.8\\target\\release\\deps -Ctarget-cpu=
native -Zorbit`
   Compiling quasi v0.11.0
     Running `rustc C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\quasi-0.11.0\\src\\lib.rs
--crate-name quasi --crate-type lib -C opt-level=3 -C metadata=f0f90e4a9a9f69ab -C extra-filename=-f0f90e4a9a9f69ab --ou
t-dir C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\serde_codegen-0.7.8\\target\\release\\de
ps --emit=dep-info,link -L dependency=C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\serde_co
degen-0.7.8\\target\\release\\deps -L dependency=C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec82
3\\serde_codegen-0.7.8\\target\\release\\deps --cap-lints allow -Ctarget-cpu=native -Zorbit`
     Running `C:\Users\Alexander\.cargo\registry\src\github.com-1ecc6299db9ec823\serde_codegen-0.7.8\target\release\buil
d\serde_codegen-da094a7fbaf20dd5\build-script-build`
   Compiling quasi_codegen v0.11.0
     Running `rustc C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\quasi_codegen-0.11.0\\src\
\lib.rs --crate-name quasi_codegen --crate-type lib -C opt-level=3 -C metadata=d15a779fc360468f -C extra-filename=-d15a7
79fc360468f --out-dir C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\serde_codegen-0.7.8\\tar
get\\release\\deps --emit=dep-info,link -L dependency=C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db
9ec823\\serde_codegen-0.7.8\\target\\release\\deps -L dependency=C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com
-1ecc6299db9ec823\\serde_codegen-0.7.8\\target\\release\\deps --extern aster=C:\\Users\\Alexander\\.cargo\\registry\\src
\\github.com-1ecc6299db9ec823\\serde_codegen-0.7.8\\target\\release\\deps\\libaster-8cc5f970a844ae51.rlib --cap-lints al
low -Ctarget-cpu=native -Zorbit`
   Compiling quasi_macros v0.11.0
     Running `rustc C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\quasi_macros-0.11.0\\src\\
lib.rs --crate-name quasi_macros --crate-type dylib -C prefer-dynamic -C opt-level=3 -C metadata=30f11f7edf6a7bdc -C ext
ra-filename=-30f11f7edf6a7bdc --out-dir C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\serde_
codegen-0.7.8\\target\\release\\deps --emit=dep-info,link -L dependency=C:\\Users\\Alexander\\.cargo\\registry\\src\\git
hub.com-1ecc6299db9ec823\\serde_codegen-0.7.8\\target\\release\\deps -L dependency=C:\\Users\\Alexander\\.cargo\\registr
y\\src\\github.com-1ecc6299db9ec823\\serde_codegen-0.7.8\\target\\release\\deps --extern quasi_codegen=C:\\Users\\Alexan
der\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\serde_codegen-0.7.8\\target\\release\\deps\\libquasi_codegen-d1
5a779fc360468f.rlib --cap-lints allow -Ctarget-cpu=native -Zorbit`
     Running `rustc src\\lib.rs --crate-name serde_codegen --crate-type lib -C opt-level=3 --cfg feature=\"nightly\" --c
fg feature=\"quasi_macros\" --out-dir C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\serde_co
degen-0.7.8\\target\\release --emit=dep-info,link -L dependency=C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-
1ecc6299db9ec823\\serde_codegen-0.7.8\\target\\release -L dependency=C:\\Users\\Alexander\\.cargo\\registry\\src\\github
.com-1ecc6299db9ec823\\serde_codegen-0.7.8\\target\\release\\deps --extern quasi_macros=C:\\Users\\Alexander\\.cargo\\re
gistry\\src\\github.com-1ecc6299db9ec823\\serde_codegen-0.7.8\\target\\release\\deps\\quasi_macros-30f11f7edf6a7bdc.dll
--extern aster=C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\serde_codegen-0.7.8\\target\\re
lease\\deps\\libaster-8cc5f970a844ae51.rlib --extern quasi=C:\\Users\\Alexander\\.cargo\\registry\\src\\github.com-1ecc6
299db9ec823\\serde_codegen-0.7.8\\target\\release\\deps\\libquasi-f0f90e4a9a9f69ab.rlib -Ctarget-cpu=native -Zorbit`
Assertion failed: isa<X>(Val) && "cast<Ty>() argument of incompatible type!", file C:\bot\slave\nightly-dist-rustc-win-m
svc-32\build\src\llvm\include\llvm/Support/Casting.h, line 223
error: Could not compile `serde_codegen`.

Caused by:
  Process didn't exit successfully: `rustc src\lib.rs --crate-name serde_codegen --crate-type lib -C opt-level=3 --cfg f
eature="nightly" --cfg feature="quasi_macros" --out-dir C:\Users\Alexander\.cargo\registry\src\github.com-1ecc6299db9ec8
23\serde_codegen-0.7.8\target\release --emit=dep-info,link -L dependency=C:\Users\Alexander\.cargo\registry\src\github.c
om-1ecc6299db9ec823\serde_codegen-0.7.8\target\release -L dependency=C:\Users\Alexander\.cargo\registry\src\github.com-1
ecc6299db9ec823\serde_codegen-0.7.8\target\release\deps --extern quasi_macros=C:\Users\Alexander\.cargo\registry\src\git
hub.com-1ecc6299db9ec823\serde_codegen-0.7.8\target\release\deps\quasi_macros-30f11f7edf6a7bdc.dll --extern aster=C:\Use
rs\Alexander\.cargo\registry\src\github.com-1ecc6299db9ec823\serde_codegen-0.7.8\target\release\deps\libaster-8cc5f970a8
44ae51.rlib --extern quasi=C:\Users\Alexander\.cargo\registry\src\github.com-1ecc6299db9ec823\serde_codegen-0.7.8\target
\release\deps\libquasi-f0f90e4a9a9f69ab.rlib -Ctarget-cpu=native -Zorbit` (exit code: 3221226505)
```

**Comment by eddyb** [comment]
@alexbool Does this still happen on the latest nightly?

**Comment by alexbool** [comment]
@eddyb latest nightly broke aster, so nope. I will recheck when it's fixed

**Comment by alexbool** [comment]
Still reproduces on latest nightly and `serde-codegen 0.7.10`

**Comment by eddyb** [comment]
@alexbool Is a debugger's backtrace in this case useless?

**Comment by alexbool** [comment]
@eddyb What do yo mean with debugger's backtrace?
Also `RUST_LOG=rustc_trans::mir` seems to do nothing.

**Comment by eddyb** [comment]
@alexbool I mean `gdb` or Visual Studio. `RUST_BACKTRACE` won't do no good because this is not a panic, it's an abort from LLVM.
`RUST_LOG` doesn't work because debug assertions are not enabled on nightly (cc @alexcrichton :cry:).

**Comment by retep998** [comment]
`gdb` won't work because it is `-msvc`. Visual Studio is the best option for getting a backtrace.

**Comment by alexbool** [comment]
So what should I precisely do? Launch `rustc` under Visual Studio and then put a breakpoint? Where? Is nightly compiler shipped with debug symbols?

**Comment by eddyb** [comment]
@alexbool It's not about breakpoints, the debugger will automatically catch the abort. The lack of debug symbols is why I asked whether it's useless.

**Comment by alexbool** [comment]
OK, I try this later today

**Comment by retep998** [comment]
Well, since rustc links to dylib versions of the standard library and friends, it actually means that those symbols which are exported from the dylibs can be found by the debugger when generating a backtrace. Granted proper debug symbols via PDB files are always better but nobody's decided on how to provide a debuginfo package for Rust yet.

**Comment by alexbool** [comment]
![default](https://cloud.githubusercontent.com/assets/1283792/15991659/3239bb44-30c2-11e6-91ee-a66ef62a8ec4.PNG)
So the backtrace looks like this...

**Comment by eddyb** [comment]
@alexbool Yeah, that's what I was afraid of :(. Thanks anyway, although I don't have a windows setup to figure this out on.
cc @nikomatsakis

**Comment by alexcrichton** [comment]
Ok, built a custom compiler with debuginfo and got a stack trace:

![https://i.imgur.com/b4LUvWH.png](https://i.imgur.com/b4LUvWH.png)

I wonder if this is fixed in upstream LLVM...

**Comment by alexcrichton** [comment]
Oh also oddly enough, for me this was faulting in syntex_syntax rather than serde_codegen, but this was also 64-bit rather than 32-bit which may have changed something.

**Comment by eddyb** [comment]
@alexcrichton That's really useful, thanks!
Judging from that stack trace, it's likely [this `cast`](https://github.com/rust-lang/llvm/blob/80ad955b60b3ac02d0462a4a65fcea597d0ebfb1/include/llvm/IR/Instructions.h#L4235) called from [inside `SimplifyCFG`](https://github.com/rust-lang/llvm/blob/80ad955b60b3ac02d0462a4a65fcea597d0ebfb1/lib/Transforms/Utils/SimplifyCFG.cpp#L3379).
LLVM's `Verifier` actually [checks that the cast would succeed](https://github.com/rust-lang/llvm/blob/80ad955b60b3ac02d0462a4a65fcea597d0ebfb1/lib/IR/Verifier.cpp#L3330-L3333), so either `SimplifyCFG` is running before the `Verifier` (in which case trans could be generating bad IR that's not checked on creation) or `SimplifyCFG` is replacing uses of `cleanuppad` with something else.

@alexbool @alexcrichton The next step would be to get the last function from the (very verbose!) log produced by compiling with `RUSTFLAGS="-Zorbit -Cllvm-args=-print-before-all"` (do not set this for any other crate than the failing one, it would probably slow everything down).

**Comment by alexbool** [comment]
@eddyb It's already compiling for 20 minutes and doesn't want to end. I see you are doing quite a lot of work in the compiler :smile: 
I'll post the results as soon they are ready

**Comment by eddyb** [comment]
@alexbool Most of it is just all of the passes that are being ran on all of the LLVM functions, so there's a lot of stuff, and it slows down LLVM, too, sadly. @alexcrichton mentioned on IRC a few hours ago that he started the same thing, but I haven't heard back since.

**Comment by alexcrichton** [comment]
@eddyb I may miss you on IRC, but I ran with `-C llvm-args=-print-before-all` yesterday and after running for ~24 hours now it's produced nearly 5GB of output and is still going. I think you can customize which passes you print before, right? Perhaps I should just try printing before the verifier pass if you think that's the one crashing?

**Comment by eddyb** [comment]
@alexcrichton That's terrible, I'm sorry.
`RUSTFLAGS="-Zorbit -Cllvm-args=-print-before=simplifycfg"` seems to print what I'm after (judging by the stack trace containing `SimplifyCFG` in it).

**Comment by alexbool** [comment]
I'll try it either.

**Comment by alexbool** [comment]
This is my log [compile.log.gz](https://github.com/rust-lang/rust/files/316086/compile.log.gz)

**Comment by eddyb** [comment]
@alexbool Thanks! Relevant part: https://gist.github.com/eddyb/ab2e6b1604897e1033f51badb4c4b9fb.

**Comment by eddyb** [comment]
So it doesn't look like that function is invalid, which means that the function might have significantly changed due to the (possibly incorrect) `SimplifyCFG` pass.

**Comment by eddyb** [comment]
@alexbool @alexcrichton Think I found the fix upstream: https://github.com/llvm-mirror/llvm/commit/47bc554474593652aa2d03b8ef047fbaea9b009a.

**Comment by alexcrichton** [comment]
Ok, @brson and I will work on https://github.com/rust-lang/rust/pull/34055, and once that's landed we can try to land an LLVM update, and hopefully that'll just magically fix this!

**Comment by alexcrichton** [comment]
I've added this to the "Launch MIR into Orbit" milestone, but feel free to correct me if that was in error!

**Comment by eddyb** [comment]
@alexbool Sounds good. What's the status of the LLVM update?

**Comment by alexcrichton** [comment]
I'm not personally actively working on it, but @brson has posted about it [here](https://internals.rust-lang.org/t/is-anyone-working-on-supporting-the-newest-llvm-version/3529/7) and [here](https://internals.rust-lang.org/t/need-help-with-emscripten-port/3154/38). We may want a little bit of extra coordinate to support both emscripten and AVR, but may just want to coordinate in [this thread](https://internals.rust-lang.org/t/is-anyone-working-on-supporting-the-newest-llvm-version/3529/7)

**Comment by nikomatsakis** [comment]
@alexcrichton seems like it qualifies for milestone to me.

**Comment by nikomatsakis** [comment]
Do we believe that this is platform specific? The original comment mentions that it fails on Windows 32-bit, but not Mac -- what about linux? Other versions of Windows?

**Comment by alexbool** [comment]
I cannot repriduce this on Linux. Did not try Windows 64 bit

**Comment by eddyb** [comment]
@alexbool Next nightly (~20h from now) should be fixed. Will close when we have confirmation.

**Comment by alexbool** [comment]
OK, I'll try

**Comment by eddyb** [comment]
@alexbool FWIW both @alexcrichton and me and have tried and can't reproduce on beta or nightly.
I also can't reproduce on a local build that includes the LLVM upgrade, so this is most likely fixed now.

**Comment by alexbool** [comment]
@eddyb I can confirm that with the latest nightly it doesn't reproduce
